### PR TITLE
CLDR-15296 Restore previous XML ordering for units

### PR DESCRIPTION
**Pull request by macchiati**
CLDR-15296

A previous change modified the DTD attribute order to be driven from the Validity data. This causes the ordering of units to change in all the locale files when CLDRModify is run. This reverts that change. It also runs CLDRModify (no args) to ensure that unit order doesn't change.

Note: In the process of fixing that, I developed code that does a regularized order that should allow use to use the same order in the Survey Tool and in XML. But that was too disruptive for now; but I want to keep the tool in the source for later refinement.

Longer term I want to restore the use of validity, instead of a hard-coded list. But we need to get this restored right away so we can use CLDRModify.

common/supplemental/units.xml
- added missing conversion exposed by tests

common/validity/unit.xml
- restored list, with new items.

tools/cldr-code/src/main/java/org/unicode/cldr/tool/UnitValidityCanonicalizer.java
- tool to use validity (eventually)

tools/cldr-code/src/main/java/org/unicode/cldr/util/DtdData.java
- restored use of hard coded list, for now

tools/cldr-code/src/main/java/org/unicode/cldr/util/Units.java
- utility for new tool

tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestUnits.java
- small fixes for test

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Comment by btangmu** [comment]
There are some compiler errors including:

Error:  /home/runner/work/cldr/cldr/tools/cldr-code/src/main/java/org/unicode/cldr/tool/UnitValidityCanonicalizer.java:[22,35] org.unicode.cldr.util.Units.TypeAndCore is not public in org.unicode.cldr.util.Units; cannot be accessed from outside package

And a formatting error, which should be fixable by running `mvn --file=tools/pom.xml spotless:apply`

I don't see any other problems.

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/UnitValidityCanonicalizer.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                                    "pressure",
+                                    "temperature",
+                                    "digital",
+                                    "short",
+                                    "angle",
+                                    "concentr",
+                                    "consumption",
+                                    "force",
+                                    "light",
+                                    "torque"))
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/tool/UnitValidityCanonicalizer.java:54]: why is this a static list? Should this go into supplemental metadata?

**Review by srl295 — APPROVED** [review]
changes seem OK, though we should be removing rather than adding static lists.

Was the "prior change" #2868 ?

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/Units.java [hunk 2]
```diff
@@ -81,6 +82,32 @@ public static String combinePattern(
...
+    }
+    /**
+     * Returns the type and core for a unit, be it long or short
+     *
+     * @param longOrShortUnit
+     * @param core
+     * @return
+     * @return
+     */
+    public static TypeAndCore splitUnit(String longOrShortUnit, TypeAndCore typeAndCore) {
```
Comment (btangmu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/Units.java:97]: is splitUnit part of determining the order of items, as in the xml? if so, ideally the order could depend less on special-purpose java methods that may be tightly coupled to many other parts of the code -- something simpler and more data-driven, and loosely coupled, might be more robust and maintainable

Comment (macchiati) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/Units.java:97]: We have unfortunately both long ids and short ids for units: length-kilometer and kilometer. The former are in our locale files, and used in the ST ordering, the latter are "real" and used in the code.
There is a ticket to drop the "length-" etc subtag, but for now we have to work with it.

**Pull request merged**